lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +124,278 @@ function EncryptUtil(encrypt, filestream, temp) {
     const keyBytes = new Buffer.from(keyBase64, BASE64);
     const ivBytes = new Buffer.from(ivBase64, BASE64);
+    const keyDecipher = crypto.createDecipheriv(AES_ECB.cipherName(decodedKek.length), decodedKek, null);
+    const fileKey = performCrypto(keyDecipher, keyBytes);
 
-    // Create temp file
-    let tempOutputFileName;
-    let tempFd;
-    await new Promise((resolve, reject) => {
-      tmp.file({ dir: tmpDir, prefix: path.basename(inFileName) + '#' }, (err, path, fd) => {
+    const dataDecipher = crypto.createDecipheriv(AES_CBC.cipherName(fileKey.length), fileKey, ivBytes);
+    return await performFileStreamCrypto(dataDecipher, tmpDir, inputFilePath, chunkSize);
+  };
+
+  //TODO: Add proper usage when feature is ready (SNOW-940981)
+  this.decryptFileGCM = async function (metadata, encryptionMaterial, inputFilePath, tmpDir = null) {
+    const keyBase64 = metadata.key;
+    const keyIvBase64 = metadata.keyIv;
+    const dataIvBase64 = metadata.iv;
+    const dataAadBase64 = metadata.dataAad;
+    const keyAadBase64 = metadata.keyAad;
+
+    const decodedKek = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
+    const keyBytes = new Buffer.from(keyBase64, BASE64);
+    const keyIvBytes = new Buffer.from(keyIvBase64, BASE64);
+    const dataIvBytes = new Buffer.from(dataIvBase64, BASE64);
+    const dataAadBytes = new Buffer.from(dataAadBase64, BASE64);
+    const keyAadBytes = new Buffer.from(keyAadBase64, BASE64);
+
+    const fileKey = this.decryptGCM(keyBytes, decodedKek, keyIvBytes, keyAadBytes);
+
+    const fileContent = await new Promise((resolve, reject) => {
+      fs.readFile(inputFilePath, (err, data) => {
         if (err) {
           reject(err);
         }
-        tempOutputFileName = path;
-        tempFd = fd;
-        resolve();
+        resolve(data);
       });
     });
 
-    // Create key decipher with decoded key and AES ECB
-    let decipher = crypto.createDecipheriv(aesEcb(keySize), decodedKey, null);
-    const fileKey = Buffer.concat([
-      decipher.update(keyBytes),
-      decipher.final()
-    ]);
-
-    // Create decipher with file key, iv bytes, and AES CBC
-    decipher = crypto.createDecipheriv(aesCbc(keySize), fileKey, ivBytes);
+    const decryptedData = this.decryptGCM(fileContent, fileKey, dataIvBytes, dataAadBytes);
+    return await writeContentToFile(tmpDir, path.basename(inputFilePath) + '#', decryptedData);
+  };
+  
+  function performCrypto(cipherOrDecipher, data) {
+    const encryptedOrDecrypted = cipherOrDecipher.update(data);
+    const final = cipherOrDecipher.final();
+    return Buffer.concat([encryptedOrDecrypted, final]);
+  }
 
+  async function performFileStreamCrypto(cipherOrDecipher, tmpDir, inputFilePath, chunkSize) {
+    let outputFilePath;
+    let outputFileFd;
+    await new Promise((resolve, reject) => {
+      tmp.file({ dir: tmpDir, prefix: path.basename(inputFilePath) + '#' }, (err, path, fd) => {
+        if (err) {
+          reject(err);
```

Comment:
after reject should be return or else - also in other places

Reply:
done